docs/architecture/grpc-for-wcf-developers/protobuf-data-types.md

```diff
@@ -1,11 +1,158 @@
+
+```protobuf  
+syntax = "proto3"
+
+import "google/protobuf/duration.proto";  
+import "google/protobuf/timestamp.proto";
+
+message Meeting {
+
+    string subject = 1;
+    google.protobuf.Timestamp time = 2;
+    google.protobuf.Duration duration = 3;
+
+}  
+```
+
+### System.Guid
+
+The `Guid` type, known as `UUID` on other platforms, is not directly supported by Protobuf and there is no well-known type for it. The best approach is to handle `Guid` values as `string` fields, using the standard `8-4-4-4-12` hexadecimal format (e.g. `45a9fda3-bd01-47a9-8460-c1cd7484b0b3`) which can be parsed by all languages and platforms. You should not use a `bytes` field for `Guid` values, as problems with endian-ness can result in erratic behavior when interacting with other platforms, such as Java.
+
+### Nullable types
+
+The Protobuf code generation for C# will use the native types, such as `int` for `int32`. This means that the values are always included and can't be null. For values that require explicit null, where you would use e.g. `int?` in your C# code, Protobuf's "Well Known Types" include wrappers that are compiled to nullable C# types. To use them, import `wrappers.proto` into your `.proto` file, like this:
+
+```protobuf  
+syntax = "proto3"
+
+import "google/protobuf/wrappers.proto"
+
+message Person {
+
+    ...
+    google.protobuf.Int32Value age = 5;
+
+}
+```
+
+Protobuf will use the simple `T?` (e.g. `int?`) for the generated message property.
+
+Here is the complete list of wrapper types with their equivalent C# type:
+
+| C# Type | Well Known Type wrapper     |
+| ------- | --------------------------- |
+| double? | google.protobuf.DoubleValue |
+| float?  | google.protobuf.FloatValue  |
+| int?    | google.protobuf.Int32Value  |
+| long?   | google.protobuf.Int64Value  |
+| uint?   | google.protobuf.UInt32Value |
+| ulong?  | google.protobuf.UInt64Value |
+
+## Decimals
+
+Protobuf doesn't natively support the .NET `decimal` type, just `double` and `float`. There is an ongoing discussion in the Protobuf project about the possibility of adding a standard `Decimal` type to the well-known types, with platform support for languages and frameworks that support it, but nothing has been implemented yet.
+
+It is possible to create a message definition to represent the `decimal` type that would work for safe serialization between .NET clients and servers, but developers on other platforms would have to understand the format being used and implement their own handling for it.
+
+### Creating a custom Decimal type for Protobuf
+
+A very simple implementation could be similar to the non-standard `Money` type used by some Google APIs, without the `currency` field.
```

Comment:
This is great. I need to push to make this improvement in the underlying protobuf C# library

Reply:
I really don't understand why there's not a Decimal already in WellKnownTypes.